enhancements/baremetal/coreos-image-in-release.md

```diff
@@ -0,0 +1,260 @@
+* Stop maintaining and shipping the non-CoreOS, RHEL-based IPA PXE files.
+* Never break existing clusters, even if they are deployed in disconnected
+  environments.
+
+### Non-Goals
+
+* Automatically switch pre-existing MachineSets to deploy with
+  `coreos-installer` instead of via QCOW2 images.
+* Update the CoreOS QCOW2 image in the cluster with each OpenShift release.
+* Provide CoreOS images for platforms other than baremetal.
+* Eliminate the extra reboot performed to update CoreOS after initial
+  provisioning.
+
+## Proposal
+
+Build a container image containing the latest CoreOS ISO and the
+`coreos-installer` binary. This container can be used e.g. as an init container
+to make the ISO available where required. The `coreos-installer iso extract
+pxe` command can also be used to produce the kernel, initrd, and rootfs from
+the ISO for the purposes of PXE booting.
+
+This image could either replace the existing content of the
+`ironic-ipa-downloader` repo, be built from the new
+`image-customization-controller` repo, or be built from a new repo.
+
+### User Stories
+
+As an operator of a disconnected cluster, I want to upgrade my cluster and have it to continue to work for provisioning baremetal machines.
+
+As an operator of an OpenShift cluster, I want to add to my cluster new
+hardware that was not fully supported in RHEL at the time I installed the
+cluster.
+
+As an operator of an OpenShift cluster, I want to ensure that the OS running on
+hosts prior to them being provisioned as part of the cluster is up to date with
+bug and security fixes.
+
+### Implementation Details/Notes/Constraints
+
+We will need to restore a [change to the Machine Config
+Operator](https://github.com/openshift/machine-config-operator/pull/1792) to
+allow working with different versions of Ignition that was [previously
+reverted](https://github.com/openshift/machine-config-operator/pull/2126) but
+should now be viable after [fixes to the
+installer](https://github.com/openshift/installer/pull/4413).
+
+The correct version of CoreOS to use is available from the openshift-installer;
+this could be obtained by using the installer container image as a build image.
+
+Because OpenShift container images are built in an offline environment, it is
+not possible to simply download the image from the public Internet at container
+build time. Instead this will be accomplished by *MAGIC*. Uploading the ISOs to
+the lookaside cache in Brew is one possibility.
+
+The RHCOS release pipeline may need to be adjusted to ensure that the latest
+ISO is always available in the necessary location.
+
+We will need to ensure that the container image is rebuilt whenever the RHCOS
+image is updated.
+
+OKD uses (rebuilt) Fedora CoreOS images instead of RHEL CoreOS, and this will
+need to be taken into account. It may be that the build environment for OKD
+allows for a more straightforward solution there (like downloading the image
+directly).
```

Comment:
@LorbusChris could you comment on how you might imagine this working for OKD?

Reply:
For OKD I think we'd just use the [fcos stream metadata](https://github.com/openshift/installer/blob/master/data/data/fcos-stream.json) there?

Reply:
Yes, I think for OKD we can just download the FCOS version pinned in the stream metadata